Formula/vpn-slice.rb

```diff
@@ -0,0 +1,24 @@
+class VpnSlice < Formula
+  include Language::Python::Virtualenv
+
+  desc "Vpnc-script replacement for easy and secure split-tunnel VPN setup"
+  homepage "https://github.com/dlenski/vpn-slice"
+  url "https://github.com/dlenski/vpn-slice/archive/v0.13.tar.gz"
+  sha256 "08f62c688b8e5f0a1f643593bd6d29c38ab92e3714a5108bfde762a3cdaf33df"
+  head "https://github.com/dlenski/vpn-slice.git"
+
+  depends_on "python"
+
+  resource "setproctitle" do
+    url "https://files.pythonhosted.org/packages/5a/0d/dc0d2234aacba6cf1a729964383e3452c52096dc695581248b548786f2b3/setproctitle-1.1.10.tar.gz"
+    sha256 "6283b7a58477dd8478fbb9e76defb37968ee4ba47b05ec1c053cb39638bd7398"
+  end
+
+  def install
+    virtualenv_install_with_resources
+  end
+
+  test do
+    assert_match "vpn-slice 0.13", shell_output("#{bin}/vpn-slice --version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
`vpn-slice` needs sudo/root credentials, so I have now replaced `--version` with a test that fails on missing credentials